File: server/src/suit.go
package main

type Suit struct {
	Name         string
	Abbreviation string   `json:"abbreviation"`
	ClueColors   []string `json:"clueColors"`
	DisplayName  string   `json:"displayName"`
	OneOfEach    bool     `json:"oneOfEach"`
	Pip          string   `json:"pip"`
	Reversed     bool     `json:"reversed"`

	AllClueColors bool `json:"allClueColors"`
	AllClueRanks  bool `json:"allClueRanks"`
	NoClueColors  bool `json:"noClueColors"`
	NoClueRanks   bool `json:"noClueRanks"`
}

File: client/src/game/types/Suit.ts
import Color from './Color';

export default interface Suit {
  readonly name: string;
  readonly abbreviation: string;
  readonly clueColors: Color[];
  readonly displayName: string;
  readonly fill: string;
  readonly fillColorblind: string;
  readonly fillColors: string[];
  readonly oneOfEach: boolean;
  readonly pip: string;
  readonly reversed: boolean;

  readonly allClueColors: boolean;
  readonly noClueColors: boolean;
  readonly allClueRanks: boolean;
  readonly noClueRanks: boolean;
}
